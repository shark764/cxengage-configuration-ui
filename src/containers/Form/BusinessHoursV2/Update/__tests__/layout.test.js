/*
 * Copyright Â© 2015-2019 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';

import BusinessHoursV2UpdateForm from '../layout';

describe('<BusinessHoursV2UpdateForm />', () => {
  it('renders update form', () => {
    const rendered = shallow(
      <BusinessHoursV2UpdateForm
        handleSubmit={() => {}}
        rules={[
          {
            name: 'rule-1',
            description: 'description-1',
            type: 'type-1',
            repeats: 'rep-1',
            every: 'ever-1',
            startDate: 'Wed Jan 15 2020 18:00:00 GMT-0600 (Central Standard Time)',
            on: {
              type: 'day',
              value: 2
            },
            hours: {
              allDay: true
            },
            id: 'id-1'
          },
          {
            name: 'rule-2',
            description: 'description-1',
            type: 'type-2',
            repeats: 'rep-2',
            every: 'ever-2',
            startDate: 'end-1',
            endDate: 'end-2',
            on: {
              type: 'day',
              value: 3
            },
            hours: {
              allDay: false
            },
            id: 'id-2'
          }
        ]}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
});
