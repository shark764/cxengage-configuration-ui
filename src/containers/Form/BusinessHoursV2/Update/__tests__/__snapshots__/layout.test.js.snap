// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BusinessHoursV2UpdateForm /> renders update form 1`] = `
<form
  onSubmit={[Function]}
>
  <styled.div>
    <styled.div>
      <Connect(DetailWrapper)
        customCaretIcon="margin-top: 6px;display: inline-block;margin-left: 23px;"
        open={true}
      >
        <Styled(DetailHeader)
          customLineSpacer="border-top: 1px solid #2E9AFE; flex-grow: 1; margin: 10px 10px 0;align-self: center;"
          fontSize="20px"
          text="Hours"
        />
        <div
          style={
            Object {
              "marginLeft": "60px",
            }
          }
        >
          <label>
            Hours and Exceptions
          </label>
          <br />
          <br />
          <FieldArray
            component={[Function]}
            name="businessRules"
          />
        </div>
      </Connect(DetailWrapper)>
    </styled.div>
  </styled.div>
</form>
`;
